master/internal/db/postgres_jobs_intg_test.go

```diff
@@ -0,0 +1,168 @@
+//go:build integration
+// +build integration
+
+package db
+
+import (
+	"testing"
+
+	"github.com/shopspring/decimal"
+	"github.com/stretchr/testify/require"
+	"gotest.tools/assert"
+
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+func TestAddJob(t *testing.T) {
+	db := setupDBForTest(t)
+
+	t.Run("add job", func(t *testing.T) {
+		jobID := model.NewJobID()
+		j := model.Job{JobID: jobID, JobType: model.JobTypeCommand}
+		err := db.AddJob(&j)
+		require.NoError(t, err)
+	})
+
+	t.Run("add job with duplicate id", func(t *testing.T) {
+		jobID := model.NewJobID()
+		j := model.Job{JobID: jobID, JobType: model.JobTypeCommand}
+		err := db.AddJob(&j)
+		require.NoError(t, err)
+
+		// change job type and re-add job
+		j.JobType = model.JobTypeExperiment
+		err = db.AddJob(&j)
+		require.Error(t, err)
+	})
+
+	t.Run("add job with no job type", func(t *testing.T) {
+		err := db.AddJob(&model.Job{JobID: model.NewJobID()})
+		require.Error(t, err)
+	})
+}
+
+func TestJobByID(t *testing.T) {
+	db := setupDBForTest(t)
+
+	t.Run("add and retrieve job", func(t *testing.T) {
+		// create and send job
+		sendJob := model.Job{
+			JobID:   model.NewJobID(),
+			JobType: model.JobTypeExperiment,
+			QPos:    decimal.NewFromInt(10),
+		}
+		err := db.AddJob(&sendJob)
+		require.NoError(t, err)
+
+		// retrieve job and test for equality
+		recvJob, err := db.JobByID(sendJob.JobID)
+		require.NoError(t, err)
+		assert.Equal(t, sendJob.JobID, recvJob.JobID)
+		assert.Equal(t, sendJob.JobType, recvJob.JobType)
+		assert.Equal(t, sendJob.OwnerID, recvJob.OwnerID)
+		assert.Equal(t, decimal.NewFromInt(10).Equal(recvJob.QPos), true)
```

Comment:
is there a reason you can't `assert.Equal(t, decimal.NewFromInt(10), recvJob.QPos)`?

Reply:
Yes, because two `decimal.Decimal` variables are not equal even if their values are equal. We need to use the `.Equal()` method to test for the values being equal.

I made a playground to demonstrate: https://go.dev/play/p/wHfBSJrVY1E